cmd/snap-confine/snap-confine-invocation.c

```diff
@@ -144,3 +145,61 @@ void sc_check_rootfs_dir(sc_invocation *inv) {
 
     die("cannot locate base snap %s", inv->base_snap_name);
 }
+
+void sc_invocation_check_homedirs(sc_invocation *inv)
+{
+    FILE *f SC_CLEANUP(sc_cleanup_file) = NULL;
+    f = fopen("var/lib/snapd/system-params", "r");
+    if (f == NULL) {
+        return;
+    }
+
+    char *line SC_CLEANUP(sc_cleanup_string) = NULL;
+    size_t line_size = 0;
+    bool config_found = false;
+    while (getline(&line, &line_size, f) != -1) {
+        if (sc_startswith(line, "homedirs=")) {
+            config_found = true;
+            break;
+        }
+    }
+
+    if (!config_found) return;
+
+    char *config_line = line + sizeof("homedirs=") - 1;
+
+    /* The homedirs setting is a comma-separated list. In order to allocate the
+     * right number of strings, let's count how many commas we have.
+     */
+    int num_commas = 0;
+    for (char *c = config_line; *c != '\0'; c++) {
+        if (*c == ',') num_commas++;
+    }
+
+    /* We add *two* elements here: one is because of course the number of
+     * actual homedirs is the number of commas plus one, and the extra one is
+     * used as an end-of-array indicator. */
+    inv->homedirs = calloc(num_commas + 2, sizeof(char *));
+    if (inv->homedirs == NULL) {
+        die("cannot allocate memory for homedirs");
+    }
+
+    // strtok_r needs a pointer to keep track of where it is in the
+    // string.
+    char *buf_saveptr = NULL;
+
+    int current_index = 0;
+    char *homedir = strtok_r(config_line, ",\n", &buf_saveptr);
+    while (homedir != NULL) {
+        if (homedir[0] == '\0') {
+            /* The typical case where we'd get an empty string is
+             * if the config file just contains
+             *
+             * homedirs=
+             */
+            continue;
+        }
+        inv->homedirs[current_index++] = sc_strdup(homedir);
+        homedir = strtok_r(NULL, "/", &buf_saveptr);
```

Comment:
I'm not sure why the delimiter is "/" here?

Reply:
It's a bug, should have been `,`